### PR TITLE
Improve byte string format decoding options

### PR DESCRIPTION
**Pull request by fxamacker**
This PR improves unreleased decoding options to be included in v2.7.0.

Changes include:
- Renamed `ByteSliceExpectedEncodingMode` to `ByteStringExpectedFormatMode`, also renamed related options
- Renamed `DecOptions.ByteSliceExpectedEncoding` to `DecOptions.ByteStringExpectedFormat`
- Added `ByteStringExpectedFormatError`, which is returned when decoding expected format fails
- Updated comments to improve docs. For example,

  ```Go
  // ByteStringExpectedFormatMode specifies how to decode CBOR byte string into Go byte slice
  // when the byte string is NOT enclosed in CBOR tag 21, 22, or 23.  An error is returned if
  // the CBOR byte string does not contain the expected format (e.g. base64) specified.
  // For tags 21-23, see "Expected Later Encoding for CBOR-to-JSON Converters"
  // in RFC 8949 Section 3.4.5.2.
  ```

### CODE REVIEW

**Comment by fxamacker** [comment]
@x448 PTAL 🙏

**Comment by fxamacker** [comment]
@benluddy PTAL :pray:

File: decode.go [hunk 1]
```diff
@@ -602,32 +634,38 @@ func (bttm ByteStringToTimeMode) valid() bool {
 	return bttm >= 0 && bttm < maxByteStringToTimeMode
 }
 
-// ByteSliceExpectedEncodingMode specifies how to decode a byte string NOT enclosed in an "expected
-// later encoding" tag (RFC 8949 Section 3.4.5.2) into a Go byte slice.
-type ByteSliceExpectedEncodingMode int
+// ByteStringExpectedFormatMode specifies how to decode CBOR byte string into Go byte slice
```
Comment (x448) [inline decode.go:637]: Makes sense.  The old suffix "EncodingMode" for a decoding mode name would have been confusing.

File: tag_test.go [hunk 2]
```diff
@@ -1503,13 +1503,13 @@ func TestEncodeBuiltinTag(t *testing.T) {
 		{
 			name: "unsigned bignum content not enclosed in expected encoding tag",
 			tag:  Tag{Number: tagNumUnsignedBignum, Content: []byte{0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00}},
-			opts: EncOptions{ByteSlice: ByteSliceExpectedEncodingBase16},
+			opts: EncOptions{ByteSlice: ByteSliceToByteStringWithExpectedConversionToBase16},
```
Comment (x448) [inline tag_test.go:1506]: It wasn't named in this PR but `ByteSliceToByteStringWithExpectedConversionToBase16` can be renamed to a shorter name.

**Review by x448 — APPROVED** [review]

**Pull request merged**